### PR TITLE
Add script's cost section

### PR DESCRIPTION
**Pull request by gingerabsurdity**

### CODE REVIEW

File: platform-features/smart-contracts.md [hunk 1]
```diff
@@ -42,5 +42,15 @@ If we plan to apply constraints on all operations for a specific asset, we canno
 * Reissue Transaction
 * Burn Transaction
 
+## Scripts' Cost
+
+For estimation of script cost, we developed an estimator class, that computing cost of scripts in nanoseconds after compilation phase. The result of compilation stage is typed abstract syntax tree (AST). The estimator class do AST traversal and compute sums for every depth: the biggest depth is an estimation of the operation's complexity. We added additional costs for `let` and `+` operators to avoid the node's overstrain.
```
Comment (vsuharnikov) [inline platform-features/smart-contracts.md]: We created benchmarks. After it, we rounded off an execution time of functions and called it `complexity`.

File: platform-features/smart-contracts.md [hunk 2]
```diff
@@ -42,5 +42,15 @@ If we plan to apply constraints on all operations for a specific asset, we canno
...
 * Burn Transaction
 
+## Scripts' Cost
+
+For estimation of script cost, we developed an estimator class, that computing cost of scripts in nanoseconds after compilation phase. The result of compilation stage is typed abstract syntax tree (AST). The estimator class do AST traversal and compute sums for every depth: the biggest depth is an estimation of the operation's complexity. We added additional costs for `let` and `+` operators to avoid the node's overstrain. 
+We found the most expensive functions:
+ - base58
+ - sigVerify
+The full results of performance tests that we conducted are presented [here](/technical-details/waves-contracts-language-description/script-performance-tests.md).
+As a result, we define the following constraint for a script cost: a script must have a size no more than `20*cost(sigVerify) ≈ 8kB`.
```
Comment (vsuharnikov) [inline platform-features/smart-contracts.md]: Same with https://github.com/wavesplatform/waves-documentation/pull/147#discussion_r193349030

Comment (gingerabsurdity) [reply platform-features/smart-contracts.md]: Done

File: platform-features/smart-contracts.md (hunk 1 above)
Comment (gingerabsurdity) [reply platform-features/smart-contracts.md]: Done

File: platform-features/smart-contracts.md [hunk 3]
```diff
@@ -42,5 +42,14 @@ If we plan to apply constraints on all operations for a specific asset, we canno
 * Reissue Transaction
 * Burn Transaction
 
+## Scripts' Cost
+
+We conducted performance tests for all aspects of our scripts. For this purpose, we developed an estimator subproject with [JMH](http://openjdk.java.net/projects/code-tools/jmh/), that computes a complexity of scripts after compilation phase by AST (Abstract Syntax Tree) traversal in special _complexity units_. _Complexity units_ is a measure of the script's relative cost: we found out the most expensive operation in terms of computational complexity and defined it equal to 100 complexity units. The most expensive functions:
```
Comment (vsuharnikov) [inline platform-features/smart-contracts.md]: hehe
> benchmark subproject

**Pull request merged**